### PR TITLE
Fix ModT errors

### PR DESCRIPTION
**Pull request by LucGenetier**
Mod(4,[1,3]) returns [0,1] in Canvas
... but it returns Error 6-11: Invalid argument type (Table). Expecting a Number value instead. Error 0-12: The function 'Mod' has some invalid arguments.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 1]
```diff
@@ -597,17 +597,39 @@ public static async ValueTask<FormulaValue> AverageTable(EvalVisitor runner, Eva
...
+            double arg0 = ((NumberValue)args[0]).Value;
+
+            if (args[1] is TableValue table)
+            {
+                List<FormulaValue> modResults = new List<FormulaValue>();
+
+                foreach (DValue<RecordValue> row in table.Rows)
+                {
+                    FormulaValue val = row.Value.GetField("Value");
+                    double arg1 = val is BlankValue ? 0d : ((NumberValue)val).Value;
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:611]: consider inverting, could be cleaner?
`val is NumberValue numVal ? numVal.Value : 0d`

**Review by lesaltzm — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 1 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: fixed

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 2]
```diff
@@ -597,17 +597,39 @@ public static async ValueTask<FormulaValue> AverageTable(EvalVisitor runner, Eva
...
+            double arg0 = ((NumberValue)args[0]).Value;
+
+            if (args[1] is TableValue table)
+            {
+                List<FormulaValue> modResults = new List<FormulaValue>();
+
+                foreach (DValue<RecordValue> row in table.Rows)
+                {
+                    FormulaValue val = row.Value.GetField("Value");
+                    double arg1 = val is NumberValue nv ? nv.Value : val is BlankValue ? 0d : throw new InvalidOperationException($"Value in the array doesn't have the expected type {val.Type._type.ToString()}");
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:611]: Should this just be CommonErrors.UnreachableCodeError?

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:611]: Let's make it simple then

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**